### ISSUE TITLE
New error - "Unterminated Template" in ram-manager.js while running autopilot.js?

### ISSUE THREAD

**Issue by cullumar** [alainbryden/bitburner-scripts#121]
![Capture](https://user-images.githubusercontent.com/73199155/166873086-c4b5f72a-f121-4f36-bc2a-aeb5670e4ab6.PNG)

Hello - after freshly running git-pull.js while in bitnode 12.8, I now receive the attached error when running autopilot.js.

Git-pull.js was run at a full stop and appears to have been successful.

This error doesn't halt autopilot.js or appear to cause any serious issues - but the error does pop repeatedly with different PIDs - approximately 1 every 30 seconds or so.

Any thoughts on what I might have done wrong?  Thanks!

**Comment by cullumar** [comment]
I believe the issue might actually be that git-pull.js is not pulling from the correct location...see double slashes in these lines of texts from git-pull.js output... (bold emphasis mine)

[02:51:34] git-pull.js: SUCCESS: Updated "stats.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main/stats.js

[02:51:34] git-pull.js: SUCCESS: Updated "stockmaster.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main/stockmaster.js

[02:51:34] git-pull.js: SUCCESS: Updated "work-for-factions.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main/work-for-factions.js

[02:51:34] git-pull.js: SUCCESS: Updated "/Flags/deleting.txt" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main**//**Flags/deleting.txt

[02:51:34] git-pull.js: SUCCESS: Updated "/Remote/grow-target.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main**//**Remote/grow-target.js

[02:51:34] git-pull.js: SUCCESS: Updated "/Remote/hack-target.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main**//**Remote/hack-target.js

[02:51:35] git-pull.js: SUCCESS: Updated "/Remote/manualhack-target.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main//Remote/manualhack-target.js

[02:51:35] git-pull.js: SUCCESS: Updated "/Remote/share.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main//Remote/share.js

[02:51:35] git-pull.js: SUCCESS: Updated "/Remote/weak-target.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main//Remote/weak-target.js

[02:51:35] git-pull.js: SUCCESS: Updated "/Tasks/backdoor-all-servers.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main//Tasks/backdoor-all-servers.js

[02:51:35] git-pull.js: SUCCESS: Updated "/Tasks/backdoor-all-servers.js.backdoor-one.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main//Tasks/backdoor-all-servers.js.backdoor-one.js

[02:51:35] git-pull.js: SUCCESS: Updated "/Tasks/contractor.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main//Tasks/contractor.js

[02:51:35] git-pull.js: SUCCESS: Updated "/Tasks/contractor.js.solver.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main//Tasks/contractor.js.solver.js

[02:51:35] git-pull.js: SUCCESS: Updated "/Tasks/crack-host.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main//Tasks/crack-host.js

[02:51:35] git-pull.js: SUCCESS: Updated "/Tasks/program-manager.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main//Tasks/program-manager.js

[02:51:35] git-pull.js: SUCCESS: Updated "/Tasks/ram-manager.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main//Tasks/ram-manager.js

[02:51:35] git-pull.js: SUCCESS: Updated "/Tasks/run-with-delay.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main//Tasks/run-with-delay.js

[02:51:35] git-pull.js: SUCCESS: Updated "/Tasks/tor-manager.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main//Tasks/tor-manager.js

[02:51:35] git-pull.js: SUCCESS: Updated "/Tasks/write-file.js" to the latest from https://raw.githubusercontent.com/alainbryden/bitburner-scripts/main//Tasks/write-file.js

**Comment by alainbryden** [comment]
In my experience, the double slash in URLs is harmless and ignored, but I fixed it nevertheless.

Could you do a `nano /Tasks/ram-manager.js` and note the contents of that file here? It seems to have been corrupted - the error message complains about line 51 but this file doesn't even have that many rows.

**Comment by cullumar** [comment]
Hi Alain - I ran a git-pull.js twice to refresh, and that eliminated the error.

Unfortunately, I didn't grab the old ram-manager.js code before I did that, so I can't paste what it was before (d'oh!) but I can confirm it did appear corrupted when I was working with it last night.  The previous code in ram-manager.js was cutting off partway through row 51 - clearly mid-statement.  I know this because I had investigated, saw the cut-off, and my workaround was to paste the current updated code into ram-manager.js.

I have no idea why it would have become corrupted that way, but I appreciate you having a look and I'm glad the error is resolved.  I think this can be closed?

Cheers!
